src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -66,6 +66,7 @@ class Container implements ResettableContainerInterface
     protected $services = array();
     protected $methodMap = array();
     protected $privates = array();
+    protected $synthetics = array();
```

Comment:
True. They need to be included in `getServiceIds` though.. however falling back to `method_exists/get_class_method` at runtime will cover.